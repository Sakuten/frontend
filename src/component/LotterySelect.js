import React from 'react';
import { observer } from 'mobx-react';

const LotterySelect = ({list, value, classroom, onChange}) => (
  <select name="lotteries" value={value} onChange={e => onChange(e.target.value)}>
      {
        list
          .filter(c => c.classroom_id === Number(classroom))
          .map(c =>
            <option value={c.id}>第{c.index + 1}公演</option>
          )
      }
  </select>
)

export default observer(LotterySelect)

