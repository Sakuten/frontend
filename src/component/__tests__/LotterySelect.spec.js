import React from 'react'
import { shallow } from 'enzyme'
import LotterySelect from '../LotterySelect'

const setup = propOverrides => {
  const props = Object.assign({
    list: [],
    value: 0,
    classroom: 0,
    onChange: jest.fn()
  }, propOverrides)

  const wrapper = shallow(<LotterySelect {...props} />)

  return {
    props,
    wrapper
  }
}

describe('components', () => {
  describe('LotterySelect', () => {
    test('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
