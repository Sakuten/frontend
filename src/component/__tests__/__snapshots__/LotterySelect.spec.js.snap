// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LotterySelect render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LotterySelect
    classroom={0}
    list={Array []}
    onChange={[Function]}
    value={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "name": "lotteries",
      "onChange": [Function],
      "value": 0,
    },
    "ref": null,
    "rendered": Array [],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "name": "lotteries",
        "onChange": [Function],
        "value": 0,
      },
      "ref": null,
      "rendered": Array [],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
